### PR TITLE
Dependecias Zudstand y React Query añadidas, ejemplo de uso de query …

### PR DESCRIPTION
**Pull request by ghost**
Created an easy example of react query in use. Called fake api: https://fakestoreapi.com/docs
Added dependencies:
Zudstand (just added, not in use)
React Query (example added, in use)

Ticket:
https://github.com/unosquare/modgen.js/issues/2

### CODE REVIEW

**Comment by AllierByron** [comment]
Ticket:
https://github.com/unosquare/modgen.js/issues/2

File: src/app/layout.tsx [hunk 1]
```diff
@@ -1,6 +1,7 @@
-import './globals.css'
+// import './globals.css'
```
Comment (oscarfgq) [inline src/app/layout.tsx]: Keep it clean. If you would need this in the future Git can help you.

File: src/app/provider.tsx [hunk 2]
```diff
@@ -0,0 +1,16 @@
+"use client";//use client es para solicionar posibles errores que las librerias no soportan en modo servidor
```
Comment (oscarfgq) [inline src/app/provider.tsx]: Keep this kind of notes for yourself.

@geoperez what about this 'use client' thing?

File: src/app/layout.tsx [hunk 3]
```diff
@@ -16,7 +17,10 @@ export default function RootLayout({
 }) {
   return (
     <html lang="en">
-      <body className={inter.className}>{children}</body>
+      <body className={inter.className}>
+        {/* hereda a todos los hijos las propiedades */}
```
Comment (oscarfgq) [inline src/app/layout.tsx]: Same as above.

File: src/app/inicio/page.tsx [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+        data.map((element:any)=>(<li key={element.id}>{element.title}</li> ))
+      }
+    </ul>
+  )
+}
+
+async function pruebaUno(){
+  //prueba sencilla de react query, fakestore api es solo una api para comprobar el funcionamiento de reactQuery
+  const data = await fetch('https://fakestoreapi.com/products?limit=5');
+  // const data = await fetch(/*url de la API*/ );
```
Comment (oscarfgq) [inline src/app/inicio/page.tsx]: ?

File: src/app/inicio/page.tsx [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+    <ul>
+      {
+        data.map((element:any)=>(<li key={element.id}>{element.title}</li> ))
+      }
+    </ul>
+  )
+}
+
+async function pruebaUno(){
+  //prueba sencilla de react query, fakestore api es solo una api para comprobar el funcionamiento de reactQuery
```
Comment (oscarfgq) [inline src/app/inicio/page.tsx]: Please remove this.

File: src/app/inicio/page.tsx [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+
+  if(status === 'error'){
+    return <p>Error</p>
+  }
+
+  console.log(data);
+  
+  return (
+
+    /**Ejemplo de abajo con la fakestoreAPI */
```
Comment (oscarfgq) [inline src/app/inicio/page.tsx]: Same as above.

File: src/app/inicio/page.tsx [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+
+  if(status === 'loading'){
+    return <p>Loading...</p> 
+  }
+
+  if(status === 'error'){
+    return <p>Error</p>
+  }
+
+  console.log(data);
```
Comment (oscarfgq) [inline src/app/inicio/page.tsx]: You won't either leave console.logs in real life code. Be careful with this.

File: src/app/provider.tsx [hunk 8]
```diff
@@ -0,0 +1,16 @@
...
+import React from 'react'
+import { QueryClient, QueryClientProvider } from 'react-query'
+
+export default function Providers({
+  children,
+}: {
+  children: React.ReactNode
+}) {
+  //boilerplate de la documentacion de react query para incializar y heredar las propiedades
```
Comment (oscarfgq) [inline src/app/provider.tsx]: Remove this too.

**Review by oscarfgq — CHANGES_REQUESTED** [review]

File: src/app/provider.tsx (hunk 2 above)
Comment (geoperez) [reply src/app/provider.tsx]: Recommended for next.js

File: .vscode/settings.json [hunk 9]
```diff
@@ -0,0 +1,3 @@
+{
```
Comment (AntonioGGovea) [inline .vscode/settings.json]: generally you don't need .vscode settings unless you want to share them with the team.
you can add .vscode to the git ignore file so you don't need to remove it from your commits every time.

**Review by AntonioGGovea — CHANGES_REQUESTED** [review]

File: tsconfig.json [hunk 10]
```diff
@@ -11,6 +15,7 @@
     "moduleResolution": "bundler",
     "resolveJsonModule": true,
     "isolatedModules": true,
+    //"jsx": "preserve",//old
```
Comment (oscarfgq) [inline tsconfig.json]: I think you're keeping it, so the comment is unnecessary.

**Review by oscarfgq — CHANGES_REQUESTED** [review]

**Review by oscarfgq — APPROVED** [review]

**Review by geoperez — APPROVED** [review]

**Review by AntonioGGovea — APPROVED** [review]

**Pull request merged**